homeassistant/components/srp_energy/sensor.py

```diff
@@ -70,98 +81,86 @@ async def async_update_data():
     )
 
     # Fetch initial data so we have data when entities subscribe
-    await coordinator.async_refresh()
+    await coordinator.async_config_entry_first_refresh()
 
-    async_add_entities([SrpEntity(coordinator)])
+    entities = []
+    for sensor_info in SENSORS_INFO:
+        sensor = SrpEnergySensor(coordinator, **sensor_info)
+        entities.append(sensor)
 
+    async_add_entities(entities)
 
-class SrpEntity(SensorEntity):
-    """Implementation of a Srp Energy Usage sensor."""
 
-    def __init__(self, coordinator):
+class SrpEnergySensor(CoordinatorEntity, SensorEntity):
+    """Implementation of a Srp Energy sensor."""
+
+    _attr_attribution = ATTRIBUTION
+
+    def __init__(
+        self,
+        coordinator,
+        name,
+        device_class,
+        unit_of_measurement,
+        source_type,
+        sensor_type,
+        icon,
+        precision,
+        state_class,
+    ):
         """Initialize the SrpEntity class."""
-        self._name = SENSOR_NAME
-        self.type = SENSOR_TYPE
-        self.coordinator = coordinator
-        self._unit_of_measurement = ENERGY_KILO_WATT_HOUR
-        self._state = None
+        super().__init__(coordinator)
+        self._name = name
+        self._device_class = device_class
+        self._unit_of_measurement = unit_of_measurement
+        self._source_type = source_type
+        self._sensor_type = sensor_type
+        self._icon = icon
+        self._precision = precision
+        self._attr_state_class = state_class
+        self._attibution = ATTRIBUTION
+
+    @property
+    def unique_id(self):
+        """Return an unique id for the sensor."""
+        return f"{DOMAIN}_{self._source_type}_{self._sensor_type}"
```

Comment:
We shouldn't change the unique_id without a migration.

We don't need to include the integration domain in the id. The entity registry is aware of the domain.